packages/compass/src/main/index.ts

```diff
@@ -97,6 +100,8 @@ async function main(): Promise<void> {
     }
     // eslint-disable-next-line @typescript-eslint/no-misused-promises
     process.on('uncaughtException', handleUncaughtException);
+    // eslint-disable-next-line @typescript-eslint/no-misused-promises
+    process.on('unhandledRejection', handleUnhandledRejection);
```

Comment:
What about exit handlers not being run when this happens? Should we open another ticket for that?

Reply:
The app is still running in this case. As we show the dialog, its part of the app. And we have [this](https://github.com/mongodb-js/compass/blob/6944c4a36ed5d1201e4b0f9b5f44dfa2e5a9868e/packages/compass/src/main/application.ts#L240) handler for that.

Reply:
This listener might not be set up by that point, it's not set up in our case where the app crashes before that happens

Reply:
followup for this: https://jira.mongodb.org/browse/COMPASS-7271